published/012-2016-August-24.md

```diff
@@ -0,0 +1,130 @@
+
+### [js-ipfs](https://github.com/ipfs/js-ipfs)
+
+js-ipfs has been moving fast. Since our workshop in Lisbon, we have established a [Roadmap](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md) for the way forward. The new roadmap has very specific milestones, which you can read about there; here is a tl:dr;
+
+- [✔️ ~~Milestone 1 - js-ipfs on the browser~~](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-1---js-ipfs-on-the-browser)
+- [Milestone 2 - Quality and correctness of the JavaScript Implementation](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-2---quality-and-correctness-of-the-javascript-implementation)
+- [Milestone 3 - Enter IPLD - Enable js-ipfs to handle both merkledag protobufs and ipld cbor objects](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-3---enter-ipld---enable-js-ipfs-to-handle-both-merkledag-protobufs-and-ipld-cbor-objects)
+- [Milestone 4 - Documentation and Developer Experience](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-4---documentation-and-developer-experience)
+- [Milestone 5 - 1st implementation Peer Routing + Content Routing (the go-ipfs DHT)](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-5---1st-implementation-peer-routing--content-routing-the-go-ipfs-dht)
+- [Milestone 6 - The last miles for complete feature parity with go-ipfs](https://github.com/ipfs/js-ipfs/blob/master/ROADMAP.md#milestone-6---the-last-miles-for-complete-feature-parity-with-go-ipfs)
+
+The [js-ipfs-api](https://github.com/ipfs/js-ipfs-api) v7 has now been shipped; a full exposition of this and the thinking behind it can be found [here](https://github.com/ipfs/js-ipfs-api/pull/305). The short story is that it now has 30% more files interfacing, which means more interop with go-ipfs and more you can do with it. For future changes, [follow this issue](https://github.com/ipfs/interface-ipfs-core/issues/55), which will communicate interface changes on js-ipfs and js-ipfs-api.
+
+Moving to pull-streams: The 🚲🏠 (bike-shedding) was strong in this one, but we've finally made a call on the Control Flow library to use in order to solve the hiccups we have had in the past with Node.js Streams. This discussion took a lot of input from several people familiar with the js-ipfs project and the contenders for the Control Flow Library. You can read the whole thread [here](https://github.com/ipfs/js-ipfs/issues/362).
+
+We are working on better changelogs for our releases, which will be automatically generated. We've finalized our commit message convention, [here](https://github.com/ipfs/community/blob/master/js-commit-guidelines.md), so that they are easily parsable and human understandable. The issue to track the development of the tooling required to add this feature to our release process is [here](https://github.com/dignifiedquire/aegir/issues/30#issuecomment-236685903).
```

Comment:
This is not that exciting for a newsletter reader, it is a 'standard expectation' for a good open source project.